ibis_substrait/compiler/translate.py

```diff
@@ -583,6 +583,36 @@ def unbound_table(
     )
 
 
+@translate.register(ops.LocalTable)
+def local_table(
+    op: ops.LocalTable, expr: ir.TableExpr, _: SubstraitCompiler, **kwargs: Any
+) -> stalg.Rel:
+    def path_type_of(name : str | None):
+        if name is None:
+            raise ValueError("local table name is None")
+        if name.startswith("file://"):
+            return "uri_folder" if name.endswith("/") else "uri_file"
```

Comment:
What should happen when `name` is a file but a user has specified a trailing `/`, and similarly when a `name` is a directory but a user hasn't specified a trailing `/`?

Reply:
I wasn't sure about this myself. IIUC, this point in the code is just a declaration, and the file (or directory) is not being opened at this time and may not even exist yet. So, there should be some way for the caller to signal a file vs a directory. My current proposal has this signal embedded within the name. A possible alternative is to add a parameter to the `local_table` operation.